### DISCUSSION TITLE
High memory usage when using table function?

### DISCUSSION THREAD

**Discussion by Cyavin** [Project-OSRM/osrm-backend#6798] [Q&A] [answered]
I'm currently working on a project that requires me to make a large distance matrix using houses from Indiana. I've previously used datasets with ~2000 locations scaled up to 13,000 to simulate weighing, but when I tried using a dataset of 12,500 actual houses, my osrm backend constantly uses too much memory and kills osrm-routed. I've allocated 10gb to my docker's WSL2 program, and am only using the indiana-latest osrm set. I was wondering if there was something specifically wrong with my backend server, as the original indiana file was only around 150mb.

**Comment by danpat** [comment, marked as answer]
I mean, a 12500x12500 table request has to calculate 156,250,000 routes.  A 2000x2000 only calculates 4,000,000 - an increase of 40x larger.  OSRM has to store all that stuff in RAM while it calculates.

Give your container more memory and try again, or split your 12500x12500 request up into several smaller requests and stitch them back together after the fact.
